### DISCUSSION TITLE
How to use TopNavigation with AppLayout in dark mode

### DISCUSSION THREAD

**Discussion by purplepenguin2** [cloudscape-design/components#3058] [Q&A] [answered]
Is there a minimal example on how to properly use the TopNavigation component with the AppLayout component in dark mode?  Here's a minimal example I have, but it doesn't have the proper bottom border like the actual demo does.  Do I need to manually style this to get it to work?  If so, is that a bug, it seems odd I would need to manually add a single 1px border when the components should otherwise work together?

I need to manually set the `borderBlockEnd` property to have a proper matching bottom border.

What currently renders:

<img width="1671" alt="image" src="https://github.com/user-attachments/assets/182b5f79-56b8-4696-b9ab-d56a628776d1">

What I believe should render:

<img width="1671" alt="image" src="https://github.com/user-attachments/assets/7206ea9a-cc0d-43e8-b354-68eed07037d8">

Minimal example code:

```jsx
import * as React from "react";
import {
  AppLayout,
  BreadcrumbGroup,
  Container,
  ContentLayout,
  Header,
  SideNavigation,
} from "@cloudscape-design/components";
import TopNavigation from "@cloudscape-design/components/top-navigation";
import Header from "@cloudscape-design/components/header";
import { applyMode, Mode } from "@cloudscape-design/global-styles";

export function App() {
  applyMode(Mode.Dark);

  return (
    <>
      <div
        id="h"
        style={
          {
            // borderBlockEnd: "solid 1px #414d5c",
          }
        }
      >
        <TopNavigation
          identity={{ href: "#", title: "Service" }}
          utilities={[
            {
              type: "button",
              text: "Link",
              href: "https://example.com/",
              external: true,
              externalIconAriaLabel: " (opens in a new tab)",
            },
            {
              type: "button",
              iconName: "notification",
              title: "Notifications",
              ariaLabel: "Notifications (unread)",
              badge: true,
              disableUtilityCollapse: false,
            },
            {
              type: "menu-dropdown",
              text: "Customer Name",
              description: "email@example.com",
              iconName: "user-profile",
              items: [
                { id: "profile", text: "Profile" },
                { id: "preferences", text: "Preferences" },
                { id: "security", text: "Security" },
                { id: "signout", text: "Sign out" },
              ],
            },
          ]}
        />
      </div>
      <AppLayout
        headerSelector="#h"
        breadcrumbs={
          <BreadcrumbGroup
            items={[
              { text: "Home", href: "#" },
              { text: "Service", href: "#" },
            ]}
          />
        }
        navigationOpen={true}
        navigation={
          <SideNavigation
            header={{
              href: "#",
              text: "Service",
            }}
            items={[{ type: "link", text: `Another Page`, href: `#` }]}
          />
        }
        content={
          <ContentLayout header={<Header variant="h1">Page header</Header>}>
            <Container header={<Header variant="h2">Container header</Header>}>
              Container content
            </Container>
          </ContentLayout>
        }
      />
    </>
  );
}

```

**Comment by cansuaa** [comment]
Hi, thanks for reaching out. It is correct that border is added in the [demo](https://cloudscape.design/examples/react/non-console.html?awsui-mode=dark), not in top navigation. The border is added in the `custom-main-header` class.

**Comment by purplepenguin2** [comment]
@cansuaa Thanks for the confirmation - could you clarify if this is technically a bug or is this intentionally not supported / requires the user to manually add the border?

Is there a best practice way to add it that would work across light/dark mode versus hard coding a color?  I guess I was hoping for a small snippet or something if there was a recommended way of doing it.

**Comment by cansuaa** [comment]
Hi, I will discuss this with the team.

**Comment by cansuaa** [comment, marked as answer]
Hi, apologies for the delayed response. I discussed with the team and it's decided to add the border to the component directly. I'll work on this.

**Reply by purplepenguin2** [reply]
@cansuaa The commit was made, reverted, reverted the revert, etc. and currently still isn't available. Is there any ETA on when this will actually be included in a release?

Related issue: https://github.com/cloudscape-design/components/issues/3230

**Reply by just-boris** [reply]
relesed: https://github.com/cloudscape-design/components/releases/tag/3.0.944
